File: programmers/카펫.py
"""
코딩테스트 연습 > 완전탐색
"""


def solution(brown, yellow):
    A = 3
    while True:
        B = (brown + 4 - 2 * A) // 2

        if A*B - 2*A - 2*B + 4 == yellow:
            return [max(A, B), min(A, B)]

        A += 1

File: programmers/카펫.kt
class Solution {
    fun solution(brown: Int, yellow: Int): IntArray {
        var totalSize = brown + yellow

        (1..totalSize).forEach { it ->
            if (totalSize % it == 0) {
                var width = totalSize / it
                var height = it

                if ((width - 2) * (height - 2) == yellow && width * 2 + height * 2 - 4 == brown) {
                    return intArrayOf(width, height)
                }
            }
        }

        return intArrayOf()
    }
}
